rareapi/views/subscription.py

```diff
@@ -25,7 +28,50 @@ def list(self, request):
             subscriptions, many=True, context={'request': request})
         return Response(serializer.data)
 
+    def create(self, request):
+        """Handle POST operations; 
+        will be used when user is on author profile and hits subscribe;
+        only info needed in request is author ID"""
 
+        follower = RareUser.objects.get(user=request.auth.user)
+        author = RareUser.objects.get(pk=request.data["author"])
+
+        subscription = Subscription()
+        subscription.follower = follower
+        subscription.author = author
+
+        if follower != author:
+            try: 
+                subscription.save()
+                serializer = SubscriptionSerializer(subscription, context={'request': request})
+                return Response(serializer.data)
+            except ValidationError as ex:
+                return Response({"reason": ex.message}, status=status.HTTP_400_BAD_REQUEST)
+        else:
+            return Response({"reason": "user cannot subscribe to their own posts"}, status=status.HTTP_400_BAD_REQUEST)
+
+    @action(methods=['get'], detail=False)
+    # Gets only active subscriptions (those without an ended_on) for the current user
+    def get_current_subscriptions(self, request):
+        subscriptions = Subscription.objects.all()
+        follower = RareUser.objects.get(user=request.auth.user)
+
+        if follower is not None:
+            subscriptions = subscriptions.filter(follower_id=follower)
+            subscriptions = subscriptions.filter(ended_on__isnull=True)
```

Comment:
this could be combined into one line by putting a comma after (follower_id=follower, ended_on__isnull=True)